Kernel/API/Syscall.h

```diff
@@ -562,6 +562,14 @@ inline uintptr_t invoke(Function function)
                  : "r"(x8)
                  : "memory");
     result = x0;
+#        elif ARCH(RISCV64)
+    register uintptr_t a0 asm("a0");
+    register uintptr_t a7 asm("a7") = function;
+    asm volatile("ecall"
+                 : "=r"(a0)
+                 : "r"(a0), "r"(a7)
```

Comment:
If I read the gcc docs correctly: <https://gcc.gnu.org/onlinedocs/gcc/extensions-to-the-c-language-family/how-to-use-inline-assembly-language-in-c-code.html#controlling-names-used-in-assembler-code>
The second `a0` should have the `"0"` constraint, and not another `"r"`, but not sure if it matters

Same would be the case for the other assembly stubs